core/src/main/scala/kafka/server/DelayedDeleteRecords.scala

```diff
@@ -74,7 +74,7 @@ class DelayedDeleteRecords(delayMs: Long,
       if (status.acksPending) {
         val (lowWatermarkReached, error, lw) = replicaManager.getPartition(topicPartition) match {
           case HostedPartition.Online(partition) =>
-            partition.leaderReplicaIfLocal match {
+            partition.leaderLogIfLocal match {
```

Comment:
Potentially something for a follow-up, but this seems racy. Presumably the partition might become a follower before the call to `lowWatermarkIfLeader`.

Reply:
Will create a separate JIRA with follow up comments from this PR.